app/components/transaction-item/transaction-item.tsx

```diff
@@ -110,28 +92,55 @@ const amountDisplayStyle = ({
   return isReceive ? styles.receive : styles.send
 }
 
-export const TransactionItem: React.FC<TransactionItemProps> = ({
-  tx,
-  navigation,
+type Props = {
+  isFirst?: boolean
+  isLast?: boolean
+  txid: string
+  subtitle?: boolean
+}
+
+export const TransactionItem: React.FC<Props> = ({
+  txid,
+  subtitle = false,
   isFirst = false,
   isLast = false,
-  subtitle = false,
-}: TransactionItemProps) => {
-  const primaryCurrency = primaryCurrencyVar()
+}) => {
+  const navigation = useNavigation<StackNavigationProp<RootStackParamList>>()
+
+  const { data: tx } = useFragment_experimental<Transaction>({
+    fragment: TransactionFragmentDoc,
+    fragmentName: "Transaction",
+    from: {
+      __typename: "Transaction",
+      id: txid,
+    },
+  })
+
   const { data: { hideBalance } = {} } = useHideBalanceQuery()
 
-  const isReceive = tx.direction === "RECEIVE"
-  const isPending = tx.status === "PENDING"
-  const description = descriptionDisplay(tx)
-  const usdAmount = computeUsdAmount(tx)
   const [txHideBalance, setTxHideBalance] = useState(hideBalance)
-  const { formatToDisplayCurrency } = useDisplayCurrency()
+  const { moneyAmountToTextWithUnits } = useDisplayCurrency()
   useEffect(() => {
     setTxHideBalance(hideBalance)
   }, [hideBalance])
 
+  if (!tx || Object.keys(tx).length === 0) {
+    return null
+  }
+
+  const isReceive = tx.direction === "RECEIVE"
+  const isPending = tx.status === "PENDING"
+  const description = descriptionDisplay(tx)
+  const walletCurrency = tx.settlementCurrency as WalletCurrency
   const pressTxAmount = () => setTxHideBalance((prev) => !prev)
 
+  const amountWithCurrency = moneyAmountToTextWithUnits({
+    amount: tx.settlementAmount,
+
+    // FIXME: will be the wrong number if non USD currency
+    currency: tx.settlementCurrency,
```

Comment:
Why will this be the wrong number?

Reply:
I want to say `if non WalletCurrency`. I'm assuming settlementCurrency in the WalletCurrency.

Reply:
Yeah I think they should always be the same